iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/parser/properties/PropertiesParseTreeVisitor.java

```diff
@@ -95,6 +95,6 @@ private static TextRange textRange(ParserRuleContext ctx) {
       ctx.start.getLine(),
       ctx.start.getCharPositionInLine(),
       ctx.stop.getLine(),
-      ctx.stop.getCharPositionInLine());
+      ctx.stop.getCharPositionInLine() + 1);
```

Comment:
Looking at the YAML tree, the end offset of tokens corresponds to the index of the next character after the last. And highlighting verifier also expects them like that. W/o this adjustment registered highlighting is by one character not enough.

Reply:
I guess it should be fixed in `PropertiesParseTreeVisitor.textRange` as we will face the same problem on issue highlighting.

Reply:
Good catch. Please add a comment on why this adjustment is needed. It means that ANTLR treats the end as the position of the last character.